sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/XmlSchemaCheck.java

```diff
@@ -223,48 +230,63 @@ public void setSchemas(String schemas) {
   }
 
   private void validate() {
-    if ("autodetect".equalsIgnoreCase(schemas)) {
-      autodetectSchemaAndValidate();
-    } else {
-      String[] schemaList = StringUtils.split(schemas, " \t\n");
-
-      validate(schemaList);
+    try {
+      if ("autodetect".equalsIgnoreCase(schemas)) {
+        autodetectSchemaAndValidate();
+      } else {
+        String[] schemaList = StringUtils.split(schemas, " \t\n");
+        validate(schemaList);
+      }
+    } catch (SchemaNotFoundException e) {
+      LOG.warn("Cannot validate file {}", getWebSourceCode());
+      LOG.warn("Cause: {}", e.toString());
     }
   }
 
   private void validate(String[] schemaList) {
-    // Create a new validator
-    Validator validator = createSchema(schemaList).newValidator();
-    setFeature(validator, Constants.XERCES_FEATURE_PREFIX + "continue-after-fatal-error", true);
-    validator.setErrorHandler(new MessageHandler());
-    validator.setResourceResolver(new SchemaResolver());
+    Schema schema = createSchema(schemaList);
+    if (schema != null) {
+      Validator validator = createSchema(schemaList).newValidator();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') NullPointerException might be thrown as 'createSchema' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2259)